### PR TITLE
Fix unconfined AppArmor profile usage for unsupported systems

### PR DESCRIPTION
**Pull request by saschagrunert**
If we select "unconfined" as AppArmor profile, then we should not error
even if the host does not support it at all. This behavior has been
fixed and a corresponding e2e test has been added as well.

Fixes https://github.com/containers/podman/issues/7545

### CODE REVIEW

File: pkg/specgen/generate/security.go [hunk 1]
```diff
@@ -60,8 +60,8 @@ func setLabelOpts(s *specgen.SpecGenerator, runtime *libpod.Runtime, pidConfig s
 func setupApparmor(s *specgen.SpecGenerator, rtc *config.Config, g *generate.Generator) error {
 	hasProfile := len(s.ApparmorProfile) > 0
 	if !apparmor.IsEnabled() {
-		if hasProfile {
-			return errors.Errorf("Apparmor profile %q specified, but Apparmor is not enabled on this system", s.ApparmorProfile)
+		if hasProfile && s.ApparmorProfile != "unconfined" {
+			return errors.Errorf("AppArmor profile %q specified, but it is not enabled on this system", s.ApparmorProfile)
```
Comment (vrothberg) [inline pkg/specgen/generate/security.go:64]: `it` refers to the profile but not to AppArmor. Changing the error message should be in a separate commit but I suggest keeping it as is.

**Review by vrothberg — COMMENTED** [review]
Besides the error message, LGTM :+1:

File: pkg/specgen/generate/security.go (hunk 1 above)
Comment (saschagrunert) [reply pkg/specgen/generate/security.go]: Yes sure, reverted that line.
